### PR TITLE
[Fix 477] Use jinja2 for templating

### LINKED ISSUE #477: Use Jinja for templating
Currently all the JS templating is done manually in Python and f-string formatting. It would be nice to use something like Jinja that was designed for this very task. As the templates get bigger, this will help us stay more understandable and maintainable.

### PR DESCRIPTION
**Pull request by PeterYusuke**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### New Feature Submission:

- [x] Does your submission pass the tests? 
- [x] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Description
Added jinja2 for templating.
Fix #477 .

#### Basic concept
Pass dictionary data from Component class to template file through jinja2.
I modified type of return value, that comes from render function in Component, from string type to dict.
And created template file with ".jinja" extentions to show data.

#### template fles
Most of js.jinja file inherited from "base_page.js.jinja" file.
"base_page.js.jinja2" has basic stracture of js file.

"pages/utils.js.jinja2" file has functions to create JSX with indenting.
Components are nested, so that I show them recursively and add indents compared with their parent's indents.
(Like, children components have 2 white space indents, grandchildren have 2 white space indents more than the children and so on.)
Condition and iterable tag are created in function.

#### template.py
I add the Environment class at the top of the file.
It has some global function "json_dumps" and "react_setter" refering pynecone project functions, 
and some global context of constant value.
These are used in template files.

#### Component class
I modified render function of Component class.
It returns dictionary data now.
To get JSX tag data, I use dict function to Tag class.
I think we don't get string of Tag class , so `__str__` function of Tag class is removed.

#### React Imports
Update import logic in "utils.py" file.
At `compile_import_statement` function, just return default and rest import from library.
When import in js file, we use like below
```js
import default from "library"
import {rest_1, test_2} from "library"
```
At  `compile_imports` function, we get list of dictionary for template engine.

#### Unit Testing
Update unit testing related above.

### CODE REVIEW

File: pynecone/components/tags/tag.py [hunk 1]
```diff
@@ -117,55 +119,22 @@ def format_prop(
         assert isinstance(prop, str), "The prop must be a string."
         return format.wrap(prop, "{", check_first=False)
 
-    def format_props(self) -> str:
+    def format_props(self) -> list:
```
Comment (ElijahAhianyo) [inline pynecone/components/tags/tag.py:122]: we should use `typing.List` to support python3.7

File: tests/components/test_tag.py [hunk 2]
```diff
@@ -126,44 +125,65 @@ def test_add_props():
...
+                "name": "box",
+                "contents": "text",
+                "props": {"color": "red", "textAlign": "center"},
+            },
         ),
     ],
 )
-def test_format_tag(tag: Tag, expected: str, windows_platform: bool):
-    """Test that the formatted tag is correct.
+def test_format_tag(tag: Tag, expected: dict):
```
Comment (ElijahAhianyo) [inline tests/components/test_tag.py:157]: use typing.Dict

**Comment by PeterYusuke** [comment]
@ElijahAhianyo 
I changed source with using typing library not only you mentioned but also other sources.

**Comment by Alek99** [comment]
Thanks @PeterYusuke we are planning on a release tomorrow. We will review and get this in the one after next Monday!

**Review by picklelo — APPROVED** [review]
@PeterYusuke awesome work!! Tested this change with the main website and it works well

**Pull request merged**